engine/src/main/java/io/camunda/zeebe/engine/processing/common/ExpressionProcessor.java

```diff
@@ -105,6 +107,29 @@ public ExpressionProcessor(
         .map(EvaluationResult::getBoolean);
   }
 
+  /**
+   * Evaluates the given expression and returns the result as boolean. If the evaluation fails or
+   * the result is not a boolean then a failure is returned.
+   *
+   * @param expression the expression to evaluate
+   * @param scopeKey the scope to load the variables from (a negative key is intended to imply an
+   *     empty variable context)
+   * @param evaluationContext the evaluation context that provide additional variables to the
+   *     VariableStateEvaluationContext
+   * @return either the evaluation result as boolean, or a failure
+   */
+  public Either<Failure, Boolean> evaluateBooleanExpression(
+      final Expression expression, final long scopeKey, final EvaluationContext evaluationContext) {
```

Comment:
🔧 Because you overload the `evaluateBooleanExpression` method, it makes sense to use this method as the implementation of the other `evaluateBooleanExpression(Expression, long)`, by passing in the `EMPTY_EVALUATION_CONTEXT`. This would remove a small bit of duplication.